identity/tests/db_queries.rs

```diff
@@ -0,0 +1,336 @@
+    db_name: String,
+}
+
+impl DbTestContext {
+    fn new(connection_url: String, db_name: String) -> Self {
+        // connect to database service
+        let conn = PgConnection::establish(&connection_url)
+            .expect("Could not connect to database service");
+
+        // create new database
+        diesel::sql_query(format!("CREATE DATABASE \"{}\"", db_name))
+            .execute(&conn)
+            .unwrap_or_else(|_| panic!("Could not create database: {}", db_name));
+
+        // reconnect to the new database
+        let conn = PgConnection::establish(&format!("{}/{}", connection_url, db_name))
+            .expect("Could not connect to database service");
+
+        // run migration
+        embedded_migrations::run(&conn).expect("Failed to apply database migration");
+
+        // insert sample data for tests
+        diesel::insert_into(users)
+            .values(&sample_data::users())
+            .execute(&conn)
+            .expect("Error inserting 'users' sample data");
+
+        diesel::insert_into(users_roles)
+            .values(&sample_data::users_roles())
+            .execute(&conn)
+            .expect("Error inserting 'users_roles' sample data");
+
+        Self {
+            connection_url,
+            db_name,
+        }
+    }
+
+    fn get_connection_url(&self) -> String {
+        format!("{}/{}", self.connection_url, self.db_name)
+    }
+}
+
+impl Drop for DbTestContext {
+    fn drop(&mut self) {
+        let conn = PgConnection::establish(&self.connection_url)
+            .expect("Could not connect to database service");
+
+        diesel::sql_query(format!(
+            "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname = '{}'",
+            self.db_name
+        ))
+        .execute(&conn)
+        .unwrap();
+
+        diesel::sql_query(format!("DROP DATABASE \"{}\"", self.db_name))
+            .execute(&conn)
+            .unwrap_or_else(|_| panic!("Could not drop database: {}", self.db_name));
+    }
+}
+
+fn get_test_configuration() -> Configuration {
+    set_var("OAUTH_CLIENT_IDENTIFIER", "test_oauth_client_identifier");
+    set_var("OAUTH_CLIENT_SECRET", "test_oauth_client_secret");
+    set_var("JWT_SECRET", "test_jwt_secret");
+
+    Configuration::init_from_env().unwrap()
+}
+
+async fn setup(
+    test_context_name: String,
+) -> Result<
+    (
+        Arc<Configuration>,
+        Arc<Pool<ConnectionManager<PgConnection>>>,
+        DbTestContext,
+    ),
+    (),
+> {
+    let configuration = Arc::new(get_test_configuration());
+    let db_test_context =
+        DbTestContext::new(configuration.db_connection_base_url(), test_context_name);
+    let db_pool = Arc::new(get_db_pool(&db_test_context.get_connection_url()));
+
+    Ok((configuration, db_pool, db_test_context))
+}
+
+// get a valid user
+#[tokio::test]
+async fn db_get_user_exists() {
+    // Arrange
+    let (_configuration, db_pool, _db_test_context) = setup(stdext::function_name!().into())
```

Comment:
If you don't need the return values you could also go with
```suggestion
    let _ = setup(stdext::function_name!().into())
```

Reply:
Well, I use the pool down below. The only thing I could do here is:
```
let (_, db_pool, _db_test_context) = setup(stdext::function_name!().into())
```